packages/playground/src/components/app-root/app-root.tsx

```diff
@@ -219,100 +219,121 @@ export class AppRoot {
       loggedUser.token as string
     );
 
-    this.updateAccount({ ...this.accountState, user: loggedUser });
+    this.updateAccount({ user: loggedUser });
 
     return loggedUser;
   }
 
-  async getBalances() {
-    const { user } = this.accountState;
+  async getBalances(): Promise<{
+    ethFreeBalanceWei: BigNumber;
+    ethMultisigBalance: BigNumber;
+  }> {
+    const {
+      user: { multisigAddress, ethAddress }
+    } = this.accountState;
     const { provider } = this.walletState;
+    const node = CounterfactualNode.getInstance();
 
-    if (!user.multisigAddress || !user.ethAddress) {
-      return;
+    if (!multisigAddress || !ethAddress) {
+      return {
+        ethFreeBalanceWei: ethers.constants.Zero,
+        ethMultisigBalance: ethers.constants.Zero
+      };
     }
 
-    const multisigBalance = parseFloat(
-      ethers.utils.formatEther(
-        (await provider.getBalance(user.multisigAddress)).toString()
-      )
-    );
+    const query = {
+      type: Node.MethodName.GET_MY_FREE_BALANCE_FOR_STATE,
+      requestId: window["uuid"](),
+      params: { multisigAddress } as Node.GetMyFreeBalanceForStateParams
+    };
 
-    const walletBalance = parseFloat(
-      ethers.utils.formatEther(
-        (await provider.getBalance(user.ethAddress)).toString()
-      )
-    );
+    const { result } = await node.call(query.type, query);
 
-    this.updateAccount({
-      ...this.accountState,
-      balance: multisigBalance,
-      accountBalance: walletBalance
-    });
+    const { balance } = result as Node.GetMyFreeBalanceForStateResult;
 
-    return {
-      balance: multisigBalance,
-      accountBalance: walletBalance
+    const vals = {
+      ethFreeBalanceWei: ethers.utils.bigNumberify(balance),
+      ethMultisigBalance: await provider!.getBalance(multisigAddress)
     };
+
+    this.accountState.updateAccount!(vals);
```

Comment:
@snario While working on `app-root`, you shouldn't call the methods on the `accountState`, since the component owns the implementations of the `update*` functions. Call directly `this.updateAccount()` to get rid of the `!`.

Reply:
ah ok